pages/product.js

```diff
@@ -8,32 +8,47 @@ module.exports = {
   sizeListSpoiler: {xpath: '//label[contains(text(),"Size")]/following-sibling::div'},
   chooseSize: {xpath: '//label[contains(text(),"Size")]/following-sibling::div/ul/li[2]'},
   addToCartButton: {xpath: '//button[@id="button-cart"]'},
-  baseProductPrice: {xpath: '//span[@class="price-new"]'},
+  baseProductPrice: {xpath: '//div[@class="price"]/span'},
 
+  async checkColourExists() {
+    return await tryTo(() => I.seeElement(this.colourListSpoiler));
+  },
   
+  async chooseProductColour() {
+    let colourPrice = 0;
+    if (await this.checkColourExists()) {
+      I.click(this.colourListSpoiler);
+      I.click(this.chooseColour);
+      colourPrice = parseFloat((await I.grabTextFrom(this.chooseColour)).match('([0-9]*[.])[0-9]*')[0]);
+    } 
+    return (colourPrice);
+
+  },  
+
+  async checkSizeExists() {
+    return await tryTo(() => I.seeElement(this.sizeListSpoiler));
+  },
 
-  chooseProductColour() {
-    I.click(this.colourListSpoiler);
-    I.click(this.chooseColour);
+  async chooseProductSize() {
```

Comment:
Your method does 2 things: selects size and gets price, which is not obvious from it's name. Also you call this method twice: in `buyProduct_test.js` to select size, then again in `getSumOfProductPrices()` to get price, which is not correct. Move instructions to get price to separate method and use it in `getSumOfProductPrices()`